### PR TITLE
[0-size Tensor No.221、155、187] Add 0-size Tensor support for paddle.nn.functional.softmax [fluid_ops]

### PR DESCRIPTION
**Pull request by co63oc**
### PR Category

Operator Mechanism

### PR Types

Bug fixes

### Description

155 paddle.nn.functional.batch_norm
修改 infermeta, 符号推导，修改为0的检查
修改kernel，x.numel() 为0返回

PaddleAPITest 测试通过，错误为torch error
image

[0-size Tensor No.221] Add 0-size Tensor support for paddle.nn.functional.softmax
221 paddle.nn.functional.softmax

softmax 增加 GPUDNN 支持 0-size
增加单测

PaddleAPITest测试通过
image

187	paddle.nn.functional.linear
XPU 补充 matmul 0-size计算，其他kernel测试通过
增加单测

PaddleAPITest测试通过，错误为torch error
image

### CODE REVIEW

**Comment by co63oc** [comment]
@DanielSun11 CI已完成需要review

**Review by DanielSun11 — APPROVED** [review]
LGTM

**Comment by DanielSun11** [comment]
修改了符号推导 请 @luotao1 协调云飞review

**Review by zyfncg — APPROVED** [review]
